packages/config/src/files.js

```diff
@@ -14,27 +15,29 @@ const handleFiles = async function ({ config: { build, ...config }, repositoryRo
   const buildDir = await getBuildDir(repositoryRoot, buildA)
   const baseRel = baseRelDir ? buildDir : repositoryRoot
   const buildB = resolvePaths(buildA, BUILD_DIR_RELATIVE_PROPS, baseRel)
-  const buildC = await addDefaultPaths(buildB, baseRel)
-  return { config: { ...config, build: buildC }, buildDir }
+  const configA = resolvePaths(config, ['functionsDirectory'], baseRel)
```

Comment:
Should we use snake case for this new property? `functions_directory`
Based on https://github.com/netlify/engineering/issues/300

Reply:
Interesting! My interpretation is that we want to use snake case for _user-facing_ fields in `netlify.toml`, but we continue to use camel case for internal fields. For example, we've recently added `node_bundler` and `external_node_modules` to `netlify.toml`, but we then translate them to `nodeBundler` and `externalNodeModules` when passing those fields to `zip-it-and-ship-it`.

Since `functionsDirectory` and `functionsDirectoryOrigin` are internal fields (i.e. they're never set directly by the user), I feel that the same criteria applies here.

What do you think?

Reply:
Yes, that makes sense.
However, what do you think about the consistency of the case of properties returned by `@netlify/config`? I am not 100% sure, but I believe it should be mostly snake_case at the moment.

Reply:
After discussing with @eduardoboucas, it turns out it is the opposite: camelCase is used more at the moment, so `functionsDirectory` makes more sense. :+1: